lib/network/modules/components/Edge.js

```diff
@@ -677,16 +677,46 @@ class Edge {
         // Ignore the orientations.
         this.labelModule.pointToSelf = true;
         var x, y;
-        var radius = this.options.selfReferenceSize;
-        if (node1.shape.width > node1.shape.height) {
+        var radius = (typeof this.options.selfReferenceSize === "number")? this.options.selfReferenceSize : this.options.selfReferenceSize.size;
+        
+        // get circle coordinates
+        if(this.options.selfReferenceSize.side !== null && typeof this.options.selfReferenceSize !== "number"){
+          if(this.options.selfReferenceSize.side === "top"){
+            //top center
+            x = node1.x;
+            y = node1.y - node1.shape.height * 0.5;            
+            point = this._pointOnCircle(x, y, radius, 0.125);
```

Comment:
Is it just me or is this the same as...

Reply:
What do you mean by the same, you mean the line which is calling _pointOnCircle if that is the case then yea it is the same but params are diff. So i guess i could just set new variable which holds diff percentage for each case and then after case sentences call it.

Reply:
Line 685 === line 705.
Line 686 === line 706.
Line 687 === line 707.
Line 688 === line 708.

Reply:
Yes, my thinking was this: i had these 4 options for position so i check these and if property is set to something weird which is not within property range [top,right,bottom,left] then i set top as default.